idb/dummy.go

```diff
@@ -516,19 +543,103 @@ type RoundUpdates struct {
 
 // Clear is used to set a RoundUpdates object back to it's default values.
 func (ru *RoundUpdates) Clear() {
-	ru.AlgoUpdates = nil
-	ru.AccountTypes = nil
-	ru.AccountDataUpdates = nil
+	//ru.AlgoUpdates = nil
+	//ru.AccountTypes = nil
+	//ru.AccountDataUpdates = nil
+	//ru.AssetUpdates = nil
+
+	ru.AlgoUpdates = make(map[[32]byte]*AlgoUpdate)
+	ru.AccountTypes = make(map[[32]byte]string)
+	ru.AccountDataUpdates = make(map[[32]byte]map[string]interface{})
 	ru.AcfgUpdates = nil
 	ru.TxnAssetUpdates = nil
-	ru.AssetUpdates = nil
+	ru.AssetUpdates = make(map[[32]byte][]AssetUpdate)
```

Comment:
I'm guessing that these were left uninitialized for performance reasons, so this is a performance vs maintainability debate.

Initializing them here would let us avoid the boilerplate nil checking in several places.

Reply:
initialized non-empty for map is probably nicer. append(nil, ...) works but maps panic if nil.